scanomatic/io/sane.py

```diff
@@ -401,7 +405,7 @@ def _select_preferred(words):
 
     def _update_mode_source(self):
 
-        default_word = SaneBase._SETTINGS_REPOSITORY[self._model][
+        default_word = SaneBase._SETTINGS_REPOSITORY[cast(str, self._model)][
```

Comment:
vad händer om self._model är None? Det blir väl ett KeyError då? Är ju i och för sig inget nytt fel isåfall. men klassen har ju en model property, borde man inte kunna typa det och använda det här.

Reply:
Den är tyvärr `Optional[str]` från `__init__`. Det är möjligt att en större refaktorering vore bättre än att slänga på `cast` här dock.

Reply:
jag tror tyvärr att i princip alla `Model` instanser e sådana att de initieras med default `None` på många fält men att de i praktiken alltid har värden... och samma sak med många argument till andra classer av samma anledning. det är nog tyvärr ett ordentligt stort jobb att arbeta bort det, även om det borde göras. Vi har en issue om att få bättre typer på modellerna #15 men kanske borde den utökas till att få bort `Optional` där möjligt med.

Reply:
om det är lätt i det här fallet att kolla om `_model` kan ändras så det alltid är en sträng så kunde det ju va intressant.

Reply:
`_model` initieras från en getter som returnerar `Optional[str]`. Det är möjligt att det går att göra så att den returnerar tom sträng istället för `None`, men det har ju eventuellt obehagliga kaskadeffekter. Ska undersöka det lite.

Reply:
Här kommer `_model` från:
```python
    @classmethod
    def _get_model(cls, model: str, logger) -> Optional[str]:
        model = model.upper()

        if model not in SaneBase._SETTINGS_REPOSITORY:

            for m, settings in SaneBase._SETTINGS_REPOSITORY.items():
                if any(
                    alias.upper() in model for alias in
                    settings[SCANNER_DATA.Aliases]
                ):
                    return m

            logger.critical(
                "Model {0} unknown, only have settings for {1}".format(
                    model,
                    list(SaneBase._SETTINGS_REPOSITORY.keys()),
                ),
            )
            return None
        return model
```

Vet inte om det är safe att byta `return None` mot `return ''`, men det känns lite tveksamt.

Reply:
näe lika bra att behålla optional isf kanske... även om det uppenbart kommer bli rätt knasigt när den används sen....